src/parser.js

```diff
@@ -1066,12 +1066,13 @@ export class Parser extends Tokenizer {
     let rhs = this.parseAssignmentExpression();
 
     this.firstExprError = null;
-    return this.markLocation(
-      operator.type === TokenType.ASSIGN
-        ? { type: "AssignmentExpression", binding: expr, expression: rhs }
-        : { type: "CompoundAssignmentExpression", binding: expr, operator: operator.type.name, expression: rhs }
-    , startLocation
-    );
+    let node = operator.type === TokenType.ASSIGN
+      ? { type: "AssignmentExpression", binding: expr, expression: rhs }
+      : { type: "CompoundAssignmentExpression", binding: expr, operator: operator.type.name, expression: rhs };
+    if (operator.type !== TokenType.ASSIGN) {
```

Comment:
I don't like doing this test twice in a row. Let's combine these two statements.